File: backend/Services/Analyzer/AnalysisResult.cs
using CareerMail.Api.Models.Enums;

namespace CareerMail.Api.Services.Analyzer;

public class AnalysisResult
{
    public bool JobRelated { get; set; }
    public string? Company { get; set; }
    public string? JobTitle { get; set; }
    public ApplicationStatus? Status { get; set; }
    public EmailClassification? Classification { get; set; }
    public string? Location { get; set; }
    public string? EmploymentType { get; set; }
    public string? Salary { get; set; }
    public string? RecruiterName { get; set; }
    public string? RecruiterEmail { get; set; }
    public string? RecruiterTitle { get; set; }
    public string? RecruiterPhone { get; set; }
    public string? RecruiterLinkedin { get; set; }
    public RecruiterType RecruiterType { get; set; } = RecruiterType.NO_RECRUITER_IDENTIFIED;
    public int? ContactConfidence { get; set; }
    public string? ContactExtractionSource { get; set; }

    public DateOnly? Deadline { get; set; }
    public DateTime? InterviewDateTime { get; set; }
    public string? InterviewType { get; set; }
    public string? InterviewLink { get; set; }
    public string? TimelineNote { get; set; }
    public double Confidence { get; set; }

    public static AnalysisResult NonJob() => new() { JobRelated = false, Confidence = 0.0 };
}

File: backend/app/services/analyzer/analysis_result.py
from dataclasses import dataclass, field
from datetime import datetime
from app.models.enums import EmailClassification, ApplicationStatus, RecruiterType


@dataclass
class RecruiterContactInfo:
    name: str | None = None
    email: str | None = None
    title: str | None = None
    phone: str | None = None
    linkedin_url: str | None = None
    recruiter_type: RecruiterType = RecruiterType.NO_RECRUITER_IDENTIFIED
    confidence: int = 0
    extraction_source: str = "Rule-based Analyzer"


@dataclass
class EmailAnalysisResult:
    is_job_related: bool = False
    classification: EmailClassification | None = None
    confidence_score: float = 0.0
    detected_company: str | None = None
    detected_role: str | None = None
    suggested_status: ApplicationStatus | None = None
    recruiter_info: RecruiterContactInfo = field(default_factory=RecruiterContactInfo)
    interview_date: datetime | None = None
    interview_type: str | None = None
    meeting_link: str | None = None
    timeline_title: str | None = None
    timeline_description: str | None = None
    notes: str | None = None
